internal/cluster/host_spec_subresource.go

```diff
@@ -110,6 +123,16 @@ func FlattenHost(host *models.Host) *map[string]interface{} {
 	if len(host.IPAddresses) > 0 && host.IPAddresses[0] != nil {
 		result["ip_address"] = host.IPAddresses[0].IPAddress
 	}
+	if len(host.PhysicalNics) > 0 {
+		var physicalNics []map[string]interface{}
+		for _, nic := range host.PhysicalNics {
+			nicMap := make(map[string]interface{})
+			nicMap["name"] = nic.DeviceName
+			nicMap["mac_address"] = nic.MacAddress
+			physicalNics = append(physicalNics, nicMap)
+		}
+		result["vmnic"] = physicalNics
```

Comment:
Can we add one line in the test to verify we're setting the "vmnic" property?

Reply:
"availability_zone_name": "",
                "host_name": "xxxxxx",
                "id": "08bf1760-a554-4d10-a665-e571d79c1f1d",
                "ip_address": "10.156.137.1",
                "license_key": "",
                "password": "",
                "serial_number": "",
                "ssh_thumbprint": "",
                "username": "",
                "vmnic": [
                  {
                    "mac_address": "00:50:56:85:03:bc",
                    "name": "vmnic0"
                  },
                  {
                    "mac_address": "00:50:56:85:0d:33",
                    "name": "vmnic1"
                  },
                  {
                    "mac_address": "00:50:56:85:a2:d2",
                    "name": "vmnic2"
                  },
                  {
                    "mac_address": "00:50:56:85:66:59",
                    "name": "vmnic3"
                  }
                ]
              },

Reply:
Hi, I ment the E2E test (here in Terraform it is called acceptance test)

Reply:
here is the results
Running tool: /usr/local/bin/go test -timeout 30s -run ^TestAccDataSourceVcfCluster$ github.com/vmware/terraform-provider-vcf/internal/provider

ok  	github.com/vmware/terraform-provider-vcf/internal/provider	0.947s

Reply:
By "fix" i mean add code to the test that checks that the above **now** works.

The idea is: write new feature, write a new test that verifies it.
In this case there should already be a test for Cluster resource import.
Indeed, if we check the file resource_cluster_test.go we can find this: https://github.com/vmware/terraform-provider-vcf/blob/fb8d57863475d8b47c56d04c4e60ec4e33959bc5/internal/provider/resource_cluster_test.go#L557

a function that test attributes on imported cluster. We need to extend the above function to check for the newly added attributes and if they are missing, it should return an error, indicating that our newly added code doesn't work.

Reply:
added it to look like this:
func TestAccDataSourceVcfCluster(t *testing.T) {
	resource.ParallelTest(t, resource.TestCase{
		PreCheck:                 func() { testAccPreCheck(t) },
		ProtoV6ProviderFactories: muxedFactories(),
		Steps: []resource.TestStep{
			{
				Config: testAccVcfClusterDataSourceConfig(
					os.Getenv(constants.VcfTestClusterDataSourceId)),
				Check: resource.ComposeTestCheckFunc(
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "name"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "domain_id"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "primary_datastore_name"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "primary_datastore_type"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "is_default"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "is_stretched"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "vds"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.0.id"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.0.host_name"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.0.ip_address"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.1.id"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.1.host_name"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.1.ip_address"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.2.id"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.2.host_name"),
					resource.TestCheckResourceAttrSet("data.vcf_cluster.cluster1", "host.2.ip_address"),
				),
			},
		},
	})
}
and getting:
Starting: /Users/burnsja/go/bin/dlv dap --listen=127.0.0.1:52527 --log-dest=3 from /Users/burnsja/Documents/GitHub/terraform-provider-vcf/internal/provider
DAP server listening at: 127.0.0.1:52527
Type 'dlv help' for list of commands.
PASS
Process 79786 has exited with status 0
Detaching

Reply:
TestAccDataSourceVcfCluster test the Cluster Datasource, not the import of the Resource Cluster.

Reply:
Take a look here: https://github.com/vmware/terraform-provider-vcf/blob/fb8d57863475d8b47c56d04c4e60ec4e33959bc5/internal/provider/resource_cluster_test.go#L95

Reply:
ok, i see now, so there is to places that where not being tested, i worked on datasource and resouce was also missing. the code is going in now.